### PR TITLE
Make items draggable in the loadout edit drawer.

### PR DESCRIPTION
**Pull request by ryan-rushton**
## Overview

This PR makes items drag and droppable in the edit loadout drawer.

I realise that we also have the double click functionality but I imagine most of our users expect to be able to drag things as it is the default way to move items in the app. TBH I didn't know I could double click until I made this change.

There is probably more refactoring I could do around some of these components too but I might leave it to this to ensure the change doesn't get too large.

https://user-images.githubusercontent.com/7344652/187911583-763b9a3b-377b-4684-8ade-a0555e1a7827.mov

### CODE REVIEW

File: src/app/dim-ui/Sheet.tsx [hunk 1]
```diff
@@ -223,12 +229,12 @@ export default function Sheet({
 
       if (
         dragHandle.current?.contains(e.target as Node) ||
-        sheetContents.current!.scrollTop === 0
+        (enableContentScrollTopDrag && sheetContents.current!.scrollTop === 0)
```
Comment (ryan-rushton) [inline src/app/dim-ui/Sheet.tsx:232]: Going to call out that I really don't like this behaviour as it isn't intuitive but it's been around forever so we shouldn't change it. I have disabled it in the loadout edit drawer as it gets a bit janky when you are dragging items inside the drawer.

File: src/app/loadout-drawer/LoadoutDrawerDropTarget.tsx [hunk 2]
```diff
@@ -49,7 +49,7 @@ export default function LoadoutDrawerDropTarget({
       canDrop: (i) =>
         itemCanBeInLoadout(i) &&
         (i.classType === DestinyClass.Unknown || classType === i.classType),
-      collect: (monitor) => ({ isOver: monitor.isOver() && monitor.canDrop() }),
+      collect: (monitor) => ({ isOver: monitor.isOver({ shallow: true }) && monitor.canDrop() }),
```
Comment (ryan-rushton) [inline src/app/loadout-drawer/LoadoutDrawerDropTarget.tsx:52]: This just ensures the outer drawer highlight doesn't happen when you are dropping a single item with the new functionality.

File: src/app/dim-ui/Sheet.tsx [hunk 3]
```diff
@@ -62,6 +62,11 @@ interface Props {
    * item popup don't close the popup they were spawned from!
    */
   allowClickThrough?: boolean;
+  /**
+   * Enables the ability click anywhere and drag the sheet down/up when the sheets contents are scrolled to the top.
+   * Defaults to true.
```
Comment (bhollis) [inline src/app/dim-ui/Sheet.tsx:67]: Hmm, not sure I'm onboard. This means that the loadout sheet can't be dragged anymore?

Comment (ryan-rushton) [reply src/app/dim-ui/Sheet.tsx:67]: I can be dragged but only by the header.

Comment (ryan-rushton) [reply src/app/dim-ui/Sheet.tsx:67]: Before this, if you were scrolled to the top of the content (or the content didn't scroll) you could click **_anywhere_** on the loadout sheet and close it. When you are dragging items inside this resulted in some funky behaviour where the sheet would move slightly when you dragged an item.

Comment (bhollis) [reply src/app/dim-ui/Sheet.tsx:67]: That's kind of what I suspected - and why I wasn't able to fix this before. That said, I'm a big fan of being able to drag the sheet to dismiss it. Is there any way to make this more targeted to just the items?

Comment (ryan-rushton) [reply src/app/dim-ui/Sheet.tsx:67]: I can have a go and see if I can tune the drop targets to be more mindful of other ones. Hopefully I will get a chance to look at it in the next couple of days.

I kind of don't like the grab sheet functionality as it is due to the conditional of being at the top of the sheet content but I imagine it is there for mobile touch scrolling? On desktop it is a bit awkward

Comment (bhollis) [reply src/app/dim-ui/Sheet.tsx:67]: Yes, for mobile scrolling. You scroll up to the top then keep pulling to pull the sheet down.

**Comment by ryan-rushton** [comment]
Closing in favour of a new PR, #9374

**Pull request closed without merging**